File: user-service/src/main/java/com/astik/user_service/client/RatingService.java
package com.astik.user_service.client;

import com.astik.user_service.dto.RatingRequest;
import com.astik.user_service.dto.RatingResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

import java.util.List;

@FeignClient(name="RATING-SERVICE")
public interface RatingService {

    @PostMapping("/api/v1/ratings")
    RatingResponse createRating(@RequestBody RatingRequest ratingRequest);

    @GetMapping("/api/v1/ratings")
    List<RatingResponse> getAllRatings();

    @GetMapping("/api/v1/ratings/users/{userId}")
    List<RatingResponse> getRatingsByUserId(@PathVariable String userId);

    @GetMapping("/api/v1/ratings/hotels/{hotelId}")
    List<RatingResponse> getRatingsByHotelId(@PathVariable String hotelId);
}

File: pythonService/rating-service/services/rating_service.py
from uuid import uuid4
from schemas.rating_request import RatingRequest
from schemas.rating_response import RatingResponse

class RatingService:
    # 👇 shared in-memory storage
    _ratings = []

    def create_rating(self, rating_request: RatingRequest) -> RatingResponse:
        rating = RatingResponse(
            id=str(uuid4()),
            user_id=rating_request.user_id,
            hotel_id=rating_request.hotel_id,
            rating=rating_request.rating,
            feedback=rating_request.feedback
        )

        self._ratings.append(rating)
        return rating

    def get_all_ratings(self):
        return self._ratings

    def get_ratings_by_user_id(self, user_id: str):
        return [r for r in self._ratings if r.user_id == user_id]

    def get_ratings_by_hotel_id(self, hotel_id: str):
        return [r for r in self._ratings if r.hotel_id == hotel_id]
